client/src/components/SandwichShow.js

```diff
@@ -98,10 +99,53 @@ const SandwichShow = (props) => {
     );
   });
 
+  const roundHalf = (num) => {
+    return Math.round(num*2)/2
+  }
+
+  let rounded = roundHalf(sandwich.averageRating)
+
+  let roundedArray = []
+  if ((rounded/1) >= 1) {
+    roundedArray.push(1)
+  } 
+  if ((rounded/2) >= 1) {
+    roundedArray.push(2)
+  }
+  if ((rounded/3) >= 1) {
+    roundedArray.push(3)
+  }
+  if ((rounded/4) >= 1) {
+    roundedArray.push(4)
+  }
+  if ((rounded/5) >= 1) {
+    roundedArray.push(5)
+  }
+  if ((rounded%1) === .5) {
+    roundedArray.push(.5)
+  }
+
+  const starIcons = roundedArray.map(starNumber => {
+    let starStyle = "fa-regular fa-star"
+    if (starNumber % 1 === 0) {
+      starStyle = "fa-solid fa-star"
+    }
+    if (starNumber === .5) {
+      starStyle = "fa-solid fa-star-half-stroke"
+    }
+    return <FontAwesomeIcon key={starNumber} icon={starStyle} />
+  })
```

Comment:
this is technically an array of reactlements so
- each Icon should be given a key
- I would prefer you make a small component called `StarIcons` that takes in a `roundedArray` as a prop, and call it here!